queue/queue.go

```diff
@@ -117,7 +117,8 @@ func (q *Queue) work() {
 
 			select {
 			case toQueue := <-q.enqueue:
-				keyPath := fmt.Sprintf("%d/%s", time.Now().UnixNano(), c.String())
+				keyPath := fmt.Sprintf("%063d/%s", q.counter, c.String())
+				q.counter++
```

Comment:
let's switch this to use the time and if the new time matches the old time just add 1 nanosecond.